MAVProxy/modules/mavproxy_DGPS.py

```diff
@@ -7,39 +7,85 @@
 from pymavlink import mavutil
 from MAVProxy.modules.lib import mp_module
 
+import socket, errno
+from pymavlink import mavutil
+from MAVProxy.modules.lib import mp_module
+
 class DGPSModule(mp_module.MPModule):
     def __init__(self, mpstate):
-        super(DGPSModule, self).__init__(mpstate, "DGPS", "DGPS injection support")
+        super(DGPSModule, self).__init__(mpstate, "DGPS", "DGPS injection support for SBP/RTCP/UBC")
         self.portnum = 13320
         self.port = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self.port.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
         self.port.bind(("127.0.0.1", self.portnum))
         mavutil.set_close_on_exec(self.port.fileno())
         self.port.setblocking(0)
-        print "Listening for DGPS packets on UDP://%s:%s" % ("127.0.0.1", self.portnum)
+        self.inject_seq_nr = 0
+        print "DGPS: Listening for RTCM packets on UDP://%s:%s" % ("127.0.0.1", self.portnum)
+    
+    def send_rtcm_msg(self, data):
+        msglen = 180;
+        
+        if (len(data) > msglen * 4):
+            print "DGPS: Message too large", len(data)
+            return
+        
+        msgs = (len(data) / msglen) + 1
```

Comment:
Can't this hit 5? (720 / 180) + 1 == 4 + 1. This would be a problem as you don't want to actually shift out a 5th element, as it could step on the first one if there was a sequence problem. With that being said I think if you just take the minimum of this equation or 4 you would be fine.